src/main/java/chalmers/pimp/controller/components/LayerItemManagerPane.java

```diff
@@ -1,57 +1,112 @@
-import javafx.scene.layout.VBox;
-import chalmers.pimp.model.canvas.layer.ILayerUpdateListener;
-import chalmers.pimp.model.canvas.layer.LayerUpdateEvent;
-import chalmers.pimp.util.Resources;
-
-/**
- * The {@code LayerItemManagerPane} class manages the layer items in the chalmers.pimp.view. Listens to the chalmers.pimp.model
- * for updates.
- */
-final class LayerItemManagerPane extends AnchorPane implements ILayerUpdateListener {
-
-  @FXML
-  private StackPane stackPane;
-
-  @FXML
-  private AnchorPane emptyLayerPane;
-
-  //Should the itemManager have real layers or just the chalmers.pimp.view aspect?
-  @FXML
-  private VBox layerItemVBox;
-
-  /**
-   * @throws IOException if the associated FXML file cannot be found.
-   */
-  LayerItemManagerPane() throws IOException {
-    ControllerUtils.makeController(this, Resources.find(getClass(), "layer_pane.fxml"));
-  }
-
-  /**
-   * The update method from the Interface {@Code} ILayerUpdateListener that updates this Managers
-   * LayerItems
-   */
-  @Override
-  public void layersUpdated(LayerUpdateEvent e) {
-    for (Node node : layerItemVBox.getChildren()) {
-      LayerItemPane layerItemPane = (LayerItemPane) node;
-      //TODO
-    }
-  }
-
-  /**
-   * Adds a LayerItemPane to this managers layerItemVBox
-   *
-   * @param layerItemPane the layerItemPane to add to the VBox for display
-   */
-  void addLayerItemPane(LayerItemPane layerItemPane) {
-    layerItemVBox.getChildren().add(layerItemPane);
-  }
-}
+package chalmers.pimp.controller.components;
+
+import chalmers.pimp.controller.ControllerUtils;
+import chalmers.pimp.model.canvas.layer.ILayerUpdateListener;
+import chalmers.pimp.model.canvas.layer.LayerUpdateEvent;
+import chalmers.pimp.util.Resources;
+import java.io.IOException;
+import javafx.fxml.FXML;
+import javafx.scene.Node;
+import javafx.scene.control.Button;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.StackPane;
+import javafx.scene.layout.VBox;
+
+/**
+ * The {@code LayerItemManagerPane} class manages the layer items in the chalmers.pimp.view. Listens
+ * to the chalmers.pimp.model for updates.
+ */
+final class LayerItemManagerPane extends AnchorPane implements ILayerUpdateListener {
+
+  @FXML
+  private StackPane stackPane;
+
+  @FXML
+  private AnchorPane emptyLayerPane;
+
+  @FXML
+  private VBox layerItemVBox;
+
+  /**
+   * @throws IOException if the associated FXML file cannot be found.
+   */
+  LayerItemManagerPane() throws IOException {
+    ControllerUtils.makeController(this, Resources.find(getClass(), "layer_pane.fxml"));
+  }
+
+  /**
+   * The update method from the Interface {@Code ILayerUpdateListener} that updates this Managers
+   * LayerItems
+   */
+  @Override
+  public void layersUpdated(LayerUpdateEvent e) {
+    for (Node node : layerItemVBox.getChildren()) {
+      LayerItemPane layerItemPane = (LayerItemPane) node;
+      if (e.getLayer() == layerItemPane.getLayer()) {
+        switch (e.getType()) {

```

Comment:
quite hard to undestand such a big method, maybe split into 2 methods? (#functional decomposition)